src/Microsoft.Health.Fhir.Azure/ExportDestinationClient/AzureExportDestinationClient.cs

```diff
@@ -104,24 +111,57 @@ public async Task CommitAsync(CancellationToken cancellationToken)
                 CloudBlockBlobWrapper blobWrapper = _uriToBlobMapping[mapping.Key.Item1];
 
                 var blockId = Convert.ToBase64String(Encoding.ASCII.GetBytes(mapping.Key.Item2.ToString("d6")));
-                uploadAndCommitTasks.Add(Task.Run(async () =>
-                {
-                    await blobWrapper.UploadBlockAsync(blockId, stream, md5Hash: null, cancellationToken);
-                    await blobWrapper.CommitBlockListAsync(cancellationToken);
 
-                    stream.Dispose();
-                }));
+                uploadTasks.Add(blobWrapper.UploadBlockAsync(blockId, stream, md5Hash: null, cancellationToken));
+                wrappersToCommit.Add(blobWrapper);
             }
 
-            await Task.WhenAll(uploadAndCommitTasks);
+            await Task.WhenAll(uploadTasks);
+
+            // Commit all the blobs that were uploaded.
+            var commitTasks = new List<Task>();
+            foreach (CloudBlockBlobWrapper wrapper in wrappersToCommit)
+            {
+                commitTasks.Add(wrapper.CommitBlockListAsync(cancellationToken));
+            }
+
+            await Task.WhenAll(commitTasks);
 
             // We can clear the stream mappings once we commit everything in memory.
+            foreach (Stream stream in _streamMappings.Values)
+            {
+                stream.Dispose();
+            }
+
             _streamMappings.Clear();
         }
 
+        public async Task InitializeForResumingExportAsync(CancellationToken cancellationToken)
+        {
+            CheckIfClientIsConnected();
+
+            IEnumerable<IListBlobItem> blobs = _blobContainer.ListBlobs();
+
+            // For each blob, dowloand the committed block list and create a corresponding CloudBlockBlobWrapper.
+            foreach (IListBlobItem blobItem in blobs)
+            {
+                CloudBlockBlob blob = blobItem as CloudBlockBlob;
+                if (blob == null)
```

Comment:
I don't know if you can make this decision. If someone creates a page blob in the location, it should not affect the export and we should just ignore and move on.

Reply:
Ya, makes sense.